canlink/utils.go

```diff
@@ -0,0 +1,54 @@
+package canlink
+
+import (
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	_binary  = 2
+	_decimal = 10
+	_hex     = 16
+)
+
+// createFile creates an *os.File given information
+func createFile(l *zap.Logger, dir string, busName string, suffix string) (*os.File, error) {
+	l.Info("createFile Entered")
+	var file *os.File
+	var builder strings.Builder
+
+	_, err := builder.WriteString(dir + "/")
+	if err != nil {
+		return &os.File{}, errors.Wrap(err, "add directory to filepath")
+	}
+
+	_, err = builder.WriteString(busName + "_")
+	if err != nil {
+		return &os.File{}, errors.Wrap(err, "add bus name to file name")
+	}
+
+	_, err = builder.WriteString(time.Now().Format(_filenameDateFormat) + "_")
+	if err != nil {
+		return &os.File{}, errors.Wrap(err, "add date to file name")
+	}
+
+	_, err = builder.WriteString(time.Now().Format(_filenameTimeFormat))
+	if err != nil {
+		return &os.File{}, errors.Wrap(err, "add time to file name")
+	}
+
+	_, err = builder.WriteString(suffix)
+	if err != nil {
+		return &os.File{}, errors.Wrap(err, "add file type to file name")
+	}
```

Comment:
Using filepath.Join(dir, fmt.Sprintf("%s_...)) would make this a lot more concise and I think is more standard